lib/node_modules/@stdlib/string/remove-last/test/test.js

```diff
@@ -128,56 +258,150 @@ tape( 'the function removes the last character of a given string (Unicode charac
 	t.end();
 });
 
-tape( 'the function returns the original string if provided zero for the second parameter', function test( t ) {
+tape( 'the function returns the original string if provided zero as the second argument', function test( t ) {
 	var out;
 
 	out = removeLast( 'hello world', 0 );
 	t.strictEqual( out, 'hello world', 'returns original string' );
 
+	out = removeLast( 'hello world', 0, {} );
+	t.strictEqual( out, 'hello world', 'returns expected value' );
+
 	t.end();
 });
 
-tape( 'the function removes the last `n` characters when supplied a second argument', function test( t ) {
+tape( 'the function removes the last `n` characters of a given string (default)', function test( t ) {
 	var out;
 
-	out = removeLast( 'hello world', 5 );
-	t.strictEqual( out, 'hello ', 'removes characters' );
+	out = removeLast( 'hello world', 1 );
+	t.strictEqual( out, 'hello worl', 'removes last character' );
+
+	out = removeLast( 'hello world', 6 );
+	t.strictEqual( out, 'hello', 'returns expected value' );
 
 	out = removeLast( '!!!', 1 );
 	t.strictEqual( out, '!!', 'removes character' );
 
 	out = removeLast( '!!!', 2 );
 	t.strictEqual( out, '!', 'removes character' );
 
+	out = removeLast( 'अनुच्छेद', 1 );
+	t.strictEqual( out, 'अनुच्छे', 'returns expected value' );
+
+	out = removeLast( '六书/六書', 1 );
+	t.strictEqual( out, '六书/六', 'returns expected value' );
+
+	out = removeLast( '🌷', 1 );
+	t.strictEqual( out, '', 'returns expected value' );
+
 	t.end();
 });
 
-tape( 'the function returns an empty string if `n` is greater than or equal to the length of the string', function test( t ) {
+tape( 'the function supports removing the last `n` characters of a provided string (mode=grapheme)', function test( t ) {
+	var opts;
 	var out;
 
-	out = removeLast( 'hello world', 12 );
-	t.strictEqual( out, '', 'returns empty string' );
+	opts = {
+		'mode': 'grapheme'
+	};
 
-	out = removeLast( '!!!', 3 );
-	t.strictEqual( out, '', 'returns empty string' );
+	out = removeLast( 'hello world', 1, opts );
+	t.strictEqual( out, 'hello worl', 'returns expected value' );
+
+	out = removeLast( 'hello world', 7, opts );
+	t.strictEqual( out, 'hell', 'returns expected value' );
+
+	out = removeLast( '!!!', 1, opts );
+	t.strictEqual( out, '!!', 'returns expected value' );
+
+	out = removeLast( '!!!', 2, opts );
+	t.strictEqual( out, '!', 'returns expected value' );
+
+	out = removeLast( 'अनुच्छेद', 1, opts );
+	t.strictEqual( out, 'अनुच्छे', 'returns expected value' );
+
+	out = removeLast( '六书/六書', 1, opts );
+	t.strictEqual( out, '六书/六', 'returns expected value' );
+
+	out = removeLast( '🌷', 1, opts );
+	t.strictEqual( out, '', 'returns expected value' );
+
+	out = removeLast( '👉🏿', 1, opts );
+	t.strictEqual( out, '👉', 'returns expected value' );
```

Comment:
This is unexpected. I would have expected an empty string here (see also https://github.com/stdlib-js/stdlib/blob/1b163e9c6c6d81533a9418206e349df3bfca3858/lib/node_modules/%40stdlib/string/remove-first/test/test.js#L329).

Can you confirm whether a non-empty string is returned?

Reply:
Following up, when I do

```
In [24]: numGraphemeClusters( '👉🏿' )
Out[24]: 1
```

in the REPL, I get the expected number of grapheme clusters. So I'd expect that `removeLast` would remove the entire cluster, not just the skin tone. It's possible that we have an off-by-one error here when computing the substring.

Reply:
I think there's an issue with `prevGraphemeClusterBreak` package, if I compare the results with `nextGraphemeClusterBreak` for a single emoji it doesn't return the same values,

```
prevGraphemeClusterBreak( '👉🏿' ) // returns 1
nextGraphemeClusterBreak( '👉🏿' ) // returns -1
```

I'll open an issue for this, because I think it is a bug. For now, I just wrote a new implementation for the `remove-first-grapheme-cluster` package that uses `nextGraphemeClusterBreak`.

Reply:
Confirmed. That does look like a bug.

Reply:
```
prevGraphemeClusterBreak( '👉' )
// returns -1
```

does return the correct value. The addition of skin tones seems to be problematic in this case.

If you could open an issue, that would be great!

Reply:
https://github.com/stdlib-js/stdlib/issues/1092 ;D